WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalManager.java

```diff
@@ -103,7 +108,8 @@ private void closeInterval(IntervalBase interval) {
 		if (interval instanceof TypingInterval) {
 			TypingInterval typingInterval = (TypingInterval) interval;
 			typingInterval.setEndingDocument(DocumentCreator
-					.createDocument(typingInterval.getEditor()));
+					.createDocument(((EditorWrapper) typingInterval
```

Comment:
I don't think so, this is the only way IMO. This is due to return argument of getEditor(); it is either Eclipse's or IntelliJ's editor, so getEditor couldn't make it to the base class.